catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java

```diff
@@ -265,6 +265,24 @@ public void cloneSpriteAndAddToStage(Sprite cloneMe) {
 			copy.look.setLookData(copy.getLookDataList().get(0));
 		}
 
+		copy.look.setPositionInUserInterfaceDimensionUnit(cloneMe.look.getXInUserInterfaceDimensionUnit(),
+				cloneMe.look.getYInUserInterfaceDimensionUnit());
+		copy.look.setTransparencyInUserInterfaceDimensionUnit(cloneMe.look.getTransparencyInUserInterfaceDimensionUnit());
+		copy.look.setColorInUserInterfaceDimensionUnit(cloneMe.look.getColorInUserInterfaceDimensionUnit());
+
+		int rotationMode = cloneMe.look.getRotationMode();
+		copy.look.setRotationMode(rotationMode);
+		if (rotationMode != Look.ROTATION_STYLE_LEFT_RIGHT_ONLY && copy.look.isFlipped()) {
+			copy.look.getLookData().getTextureRegion().flip(true, false);
+			copy.look.setFlipped(false);
+		}
+		boolean orientedLeft = copy.look.getDirectionInUserInterfaceDimensionUnit() < 0;
+		if (rotationMode == Look.ROTATION_STYLE_LEFT_RIGHT_ONLY && orientedLeft) {
+			copy.look.getLookData().getTextureRegion().flip(true, false);
+			copy.look.setFlipped(true);
+		}
+		copy.look.setDirectionInUserInterfaceDimensionUnit(cloneMe.look.getDirectionInUserInterfaceDimensionUnit());
+
```

Comment:
Good catch!
The biggest flaw in the current implementation is, that the Look needs a reference to the Sprite, which isn't necessary at all. Currently it's used for

* `sprite.isPaused` → Either use [Actor.setTouchable](https://libgdx.badlogicgames.com/nightlies/docs/api/com/badlogic/gdx/scenes/scene2d/Actor.html#setTouchable-com.badlogic.gdx.scenes.scene2d.Touchable-) or move the isPaused member to the Look. It would be even better to disable touch events on the whole stage.
* `sprite.createWhenScriptActionSequence` → The Sprite should register a listener on the Look, which for simplicity just calls `createWhenScriptActionSequence`

Unfortunately a simple `Look.clone()` method still can't be created. The PhysicsLook needs a fresh PhysicsObject as parameter when created... If you come up with a solution for this, let me know. :smile_cat:

> In other words, is it OK to remove this line?

The method is used to create a Look or PhysicsLook depending on the used Bricks. Is it possible to move the method to `cloneForCloneBrick`? It's important that `resetSprite` is called __after__ the bricks have been cloned. Otherwise the wrong Look will be used.

1. clone bricks
2. call `resetSprite`
3. call `Look.copyLookForSprite` (also override this method in the PhysicsLook to clone friction, bounce factor, etc).

Hopefully this will work!? :hushed: